ocaml/xapi/xapi_sr_operations.ml

```diff
@@ -206,15 +206,6 @@ let update_allowed_operations ~__context ~self : unit =
   let keys = Hashtbl.fold (fun k v acc -> if v = None then k :: acc else acc) valid [] in
   Db.SR.set_allowed_operations ~__context ~self ~value:keys
 
-(** Someone is cancelling a task so remove it from the current_operations *)
-let cancel_task ~__context ~self ~task_id =
-  let all = List.map fst (Db.SR.get_current_operations ~__context ~self) in
-  if List.mem task_id all then
-    begin
-      Db.SR.remove_from_current_operations ~__context ~self ~key:task_id;
-      update_allowed_operations ~__context ~self
```

Comment:
I wonder why we were calling `update_allowed_operations` here but not in `cancel_tasks` below this function...?

Reply:
Because this `cancel_tasks` functions only called in `Cancel_tasks.cancel_tasks_on_host` and in the end of the function there are a separate function to handle the `update_allowed_operations` stuff.